File: Project_Scala/src/main/scala/Grammar.scala
import scala.collection.mutable.ListBuffer
import scala.io.Source

class Grammar(var terminals : List[String], var nonTerminals : List[String], var rules : List[Rule]) {
  def readGrammar(inputUrl : String): Unit = {
    val source = Source.fromFile(inputUrl)
    val lines = source.getLines()
    nonTerminals = lines.next().split(" ").toList
    terminals = lines.next().split(" ").toList
    terminals = terminals :+ "$"

    val rulesB = new ListBuffer[Rule]()
    while (lines.hasNext) {
      val line = lines.next()
      rulesB += new Rule(line.split("->")(0), line.split("->")(1).split("").toList, rulesB.length)
    }
    rules = rulesB.toList
    source.close()
  }
}

File: Project_Python/src/Grammar.py
class Grammar:
    nonTerminals = []
    terminals = []

    def __init__(self, inputUrl, rules):
        f = open(inputUrl, 'r')

        self.nonTerminals.append("R")
        self.nonTerminals.appendAll(f.readline().split(" "))

        self.terminals.appendAll(f.readline().split(" "))
        self.terminals.append("$")

        line = f.readline()
        while line:
            rule = (line.split("→")[0], line.split("→")[1].split(""), rules.len())
            rules.add(rule)
            line = f.readline()
        f.close()
